### DISCUSSION TITLE
Concat Warnings and errors at end

### DISCUSSION THREAD

**Discussion by dicko2** [cake-build/cake#3538] [Ideas]
A few of the cake scripts we have are very long, and during the build there may be several warnings or errors outputted.

Similar to msbuild, i think it would be good to have an option to be able to output at the end of the script any errors (or warnings) logged during the running of this script. 

i.e. from calls to the methods Logging.Error and Logging.Warning, perhaps a new method .OutputAllWarningsAndErrors() and/or .OutputAllErrors() that you could optionally call at the end of your script, or a public bool property you can set that will make an action fire in the teardown, I think there is several ways you could implement it.

Here is one example in recent days of a request for this. https://github.com/nunit/nunit/pull/1332#issuecomment-194779977 second last line in this comment

Currently I've been coding my way around it to provide a summary at the end, which isn't too hard, but considering its a built in feature to msbuild and its a not uncommon request I think it would be worth building in to cake.

**Comment by jnm2** [comment]
Came here to suggest the same thing. I've been missing warnings. I like it the way test runners do it where they specify the count of the warnings and repeat them so that you don't have to scroll and use the search feature to find them.

**Comment by devlead** [comment]
This would likely be an pretty easy thing now to do in a module proxying ICakeLog as a PoC.
